src/vs/base/browser/ui/grid/grid.ts

```diff
@@ -728,6 +835,7 @@ export interface ISerializedGrid {
 	orientation: Orientation;
 	width: number;
 	height: number;
+	maximizedView?: GridLocation | undefined;
```

Comment:
This is how serialized grids gets saved in memory, so we can (and should) optimize the data structure such that it minimizes room for error. What if `maximizedView` is a bogus location?

Why not make `ISerializedLeafNode` have a `maximized?: boolean` property instead? The contract would be then that, when the `SerializedGrid` deserialized, it would maximize the first leaf node which has that property set to true.

Reply:
It's just a bit weird that the maximized view is stored in the grid directly and when serialized, in the node. It makes it harder to serialize/desirialize